lib/assert.js

```diff
@@ -931,6 +932,88 @@ assert.doesNotMatch = function doesNotMatch(string, regexp, message) {
   internalMatch(string, regexp, message, doesNotMatch);
 };
 
+const noop = () => {};
+
+const mustCallChecks = [];
+
+function runCallChecks(exitCode) {
+  if (exitCode !== 0) return;
+
+  const failed = mustCallChecks.filter((context) => {
+    if (typeof context.minimum === 'number') {
+      context.messageSegment = `at least ${context.minimum}`;
+      return context.actual < context.minimum;
+    } else {
+      context.messageSegment = `exactly ${context.exact}`;
+      return context.actual !== context.exact;
+    }
+  });
+
+  failed.forEach(function(context) {
+    console.error('Mismatched %s function calls. Expected %s, actual %d.',
+                  context.name,
+                  context.messageSegment,
+                  context.actual);
+    console.error(context.stack.split('\n').slice(2).join('\n'));
+  });
+
+  if (failed.length) process.exit(1);
+}
+
+assert.mustCall = function mustCall(fn, exact) {
+  return _mustCallInner(fn, exact, 'exact');
+};
+
+assert.mustCallAtLeast = function mustCallAtLeast(fn, minimum) {
+  return _mustCallInner(fn, minimum, 'minimum');
+};
+
+function _mustCallInner(fn, criteria = 1, field) {
+  if (process._exiting)
+    // eslint-disable-next-line no-restricted-syntax
+    throw new Error('Cannot use common.mustCall*() in process exit handler');
```

Comment:
We can (should, I think) use one of our custom errors for this. We might need to add a new code for it.

Reply:
I added a new error just there